1-development/list_posts_by_rating/src/main/java/com/mixfar/posts/controllers/PostController.java

```diff
@@ -0,0 +1,111 @@
+package com.mixfar.posts.controllers;
+
+import com.mixfar.posts.model.Post;
+import com.mixfar.posts.service.PostService;
+import com.mixfar.posts.utils.Utils;
+import com.mixfar.posts.utils.enums.ErrorCode;
+import com.mixfar.posts.utils.errors.ErrorMessage;
+import com.mixfar.posts.utils.errors.FieldErrorMessage;
+import com.mixfar.posts.utils.response.DataResponse;
+import com.mixfar.posts.utils.response.ErrorResponse;
+import com.mixfar.posts.utils.response.Response;
+import com.mixfar.posts.validations.PostValidation;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@RestController
+public class PostController {
+
+    @Autowired
+    private PostService postService;
+
+    @RequestMapping(value = "/create", method = RequestMethod.POST)
```

Comment:
Why did you use this endpoint to create posts? Is there any better option?

Reply:
Since the request is a `POST` request, the endpoint can be only `/post` because we know it's for creating posts due to the request being a `POST`